digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/dao/csvexporter/AssetReporterDAO.scala

```diff
@@ -41,23 +41,58 @@ class AssetReporterDAO {
     }
   }
 
+  def laneQuery(linkIds: Seq[Long]): List[AssetReport] = {
+    MassQuery.withIds(linkIds.toSet){ idTableName =>
+      sql"""SELECT ${Lanes.typeId}, ${Lanes.nameFI}, ${Lanes.geometryType}, l.MODIFIED_BY, l.MODIFIED_DATE
+      FROM LANE l
+        JOIN LANE_LINK ll ON l.id = ll.lane_id
+        JOIN LANE_POSITION pos ON ll.lane_position_id = pos.id
+        JOIN #$idTableName i ON i.id = pos.link_id
+      """.as[AssetReport](getResult).list
+    }
+  }
 
-  def pointAssetQuery(municipalities: Seq[Long], assetTypesList: Seq[Int] ): List[AssetReport] = {
+  def manoeuvreQuery(linkIds: Seq[Long]): List[AssetReport] = {
+    MassQuery.withIds(linkIds.toSet){ idTableName =>
+      sql"""SELECT ${Manoeuvres.typeId}, ${Manoeuvres.nameFI}, ${Manoeuvres.geometryType}, m.MODIFIED_BY, m.MODIFIED_DATE
+      FROM MANOEUVRE m
+      WHERE (m.valid_to IS NULL OR m.valid_to > SYSDATE)
+      AND EXISTS
+        (SELECT * FROM MANOEUVRE_ELEMENT me
+          JOIN #$idTableName i ON i.id = me.link_id OR i.id = me.dest_link_id
+        WHERE me.manoeuvre_id = m.id)
+      """.as[AssetReport](getResult).list
+    }
+  }
+
+  def pointAssetQuery(linkIds: Seq[Long], assetTypesList: Seq[Int]): List[AssetReport] = {
     val filter = if (assetTypesList.isEmpty) "" else s"AND a.asset_type_id IN (${assetTypesList.mkString(",")})"
 
-    MassQuery.withIds(municipalities.toSet){idTableName =>
+    MassQuery.withIds(linkIds.toSet){ idTableName =>
       sql"""SELECT a.ASSET_TYPE_ID, at2.NAME, at2.GEOMETRY_TYPE, a.MODIFIED_BY, a.MODIFIED_DATE
       FROM ASSET a
         JOIN ASSET_TYPE at2 ON at2.id = a.asset_type_id
-        JOIN #$idTableName i ON i.id = a.municipality_code
+        JOIN ASSET_LINK al ON a.id = al.asset_id
+        JOIN LRM_POSITION pos ON al.position_id = pos.id
+        JOIN #$idTableName i ON i.id = pos.link_id
       WHERE (a.valid_to IS NULL OR a.valid_to > SYSDATE)
       AND at2.GEOMETRY_TYPE = 'point'
       #$filter
       """.as[AssetReport](getResult).list
     }
-
   }
```

Comment:
In this method, we need to remove the if where it's checked if assetTypesList is empty or not because they are mandatory, at least one asset type.
We can put the ELSE of that verification directly at the code without any problem.

Reply:
Applied